src/sensor/protocoldetector.cpp

```diff
@@ -190,18 +190,26 @@ bool ProtocolDetector::checkBuffer(const QByteArray& buffer)
     // since the buffer need to be clear for each try
     PingParserExt parser;
 
-    // Print information from detected devices
-    connect(&parser, &Parser::newMessage, this, [&](const ping_message& msg) {
-        common_device_information device_information(msg);
-        qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << "Detect new device:"
-                                                << "\ndevice_revision:" << device_information.device_revision()
-                                                << "\nfirmware_version_major:" << device_information.firmware_version_major()
-                                                << "\nfirmware_version_minor:" << device_information.firmware_version_minor()
-                                                << "\nfirmware_version_patch:" << device_information.firmware_version_patch();
-    });
-
+    qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << buffer;
     for (const auto& byte : buffer) {
         if(parser.parseByte(byte) == Parser::NEW_MESSAGE) {
+            // Print information from detected devices
+            common_device_information device_information(parser.rxMessage());
+            qCDebug(PING_PROTOCOL_PROTOCOLDETECTOR) << "Detect new device:"
+                                                    << "\ndevice_type:" << device_information.device_type()
+                                                    << "\ndevice_revision:" << device_information.device_revision()
+                                                    << "\nfirmware_version_major:" << device_information.firmware_version_major()
+                                                    << "\nfirmware_version_minor:" << device_information.firmware_version_minor()
+                                                    << "\nfirmware_version_patch:" << device_information.firmware_version_patch();
+
+            //TODO: Ping1D with firmware 3.26 or older
+            // We should remove this code in future releases but allowing undetected devices to be flashed
+            PingDeviceType type = static_cast<PingDeviceType>(device_information.device_type());
+            if(type == PingDeviceType::PING360) {
```

Comment:
IMO, this is an example where implicit conversions are desirable (less code), but we can't with `enum class`.

Reply:
This will be fixed after moving ping-message class to enum class ;)

Reply:
I think you missed my point: 

> IMO, this is an example where implicit conversions are desirable (less code), but **we can't with enum class.**

I don't like `enum class` right here, it is not the right place for `enum class`.

IMO:
`enum class` is good to use sometimes, but not here. In this particular situation, implicit conversions are **desireable** and better. `enum class` does not allow implicit conversions. This is part of the reason I have been recommending against using it for the ping-message api.

Reply:
Implicit conversion is only necessary because ping-message does not provide the enums. The problem is in ping-message and not ping-viewer. Implicit conversions are necessary in pingviewer because ping-message does not. It's not safe to do this kind of conversions, is not save for the programmer to do this kind of conversions, we should do it in our low level API not here. Yes, right now we need to do implicit conversions, but this is an API problem in ping-message and not pingviewer. If you dislike implicit conversions, this should be fixed in ping-message providing the correct output as enum classes.

Reply:
> If you dislike implicit conversions

I **do like** the **implicit** conversions. That is my point. The implicit conversion is safe in this case because message id is an integer **by definition**.